synapse/axon.py

```diff
@@ -33,7 +33,14 @@
 MAX_SPOOL_SIZE = CHUNK_SIZE * 32  # 512 mebibytes
 MAX_HTTP_UPLOAD_SIZE = 4 * s_const.tebibyte
 
-class AxonHttpUploadV1(s_httpapi.StreamHandler):
+class AxonHandlerMixin:
+    def getAxon(self):
```

Comment:
Would we want to make this async so that a remote axon ( via telepath Client ) could be awaited for ready status?

Reply:
We can do that; I was using the existing pattern we have for the Storm handlers which is a sync call. An implementation that references a telepath Client would implicitly have a waitready() call in it, so we may be doing double duty there.

Reply:
Is this a separate story or still relevant to this PR?

Reply:
@Cisphyx I think that would be seperate?